### PR TITLE
Add LXC built in support deprecation notice

### PR DESCRIPTION
**Pull request by crosbymichael**
Signed-off-by: Michael Crosby crosbymichael@gmail.com

### CODE REVIEW

File: docs/misc/deprecated.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ parent = "mn_use_docker"
 
 The following list of features are deprecated.
 
+### LXC built-in exec driver
+**Target For Removal In Release: v1.9**
+
+The built in LXC execution driver will be deprecated soon for an external implementation.
```
Comment (tiborvass) [inline docs/misc/deprecated.md]: `built-in`

**Comment by calavera** [comment]
Should we say `IS deprecated and will be removed in docker 1.10` now that we have deprecation guidelines in place?

**Comment by thaJeztah** [comment]
+1 on @calavera's suggestion. Are we still able to have lxc in 1.9 though? Don't know the exact plans

File: docs/misc/deprecated.md [hunk 2]
```diff
@@ -12,6 +12,12 @@ parent = "mn_use_docker"
 
 The following list of features are deprecated.
 
+### LXC built-in exec driver
+**Target For Removal In Release: v1.9**
```
Comment (duglin) [inline docs/misc/deprecated.md]: Is there a "Deprecated In Release" date we can add?  Maybe 1.8 ?

File: daemon/execdriver/execdrivers/execdrivers_linux.go [hunk 3]
```diff
@@ -19,6 +20,7 @@ func NewDriver(name string, options []string, root, libPath, initPath string, sy
 		// we want to give the lxc driver the full docker root because it needs
 		// to access and write config and template files in /var/lib/docker/containers/*
 		// to be backwards compatible
+		logrus.Warn("LXC built-in support will be deprecated soon.")
```
Comment (duglin) [inline daemon/execdriver/execdrivers/execdrivers_linux.go]: s/will be deprecated soon/is deprecated/
No point in teasing.

File: docs/misc/deprecated.md (hunk 1 above)
Comment (duglin) [inline docs/misc/deprecated.md]: s/will be deprecated soon/is deprecated/

**Comment by duglin** [comment]
+1 to @calavera  - it "is deprecated" now, but "will be removed in release 1.x"

**Comment by calavera** [comment]
LGTM

**Comment by duglin** [comment]
LGTM moving to docs review

**Comment by thaJeztah** [comment]
Hm, wanted to say the "deprecation policy" mentions two releases, but it looks like it's a bit open to interpretation;

> Before an existing feature is removed it will be labeled as "deprecated" ... and will remain in Docker for, usually, at least 2 releases.

LGTM then, I guess (but I would have preferred giving users one more release to adapt)

Do we have statistics, i.e., how many users are still running with the LXC driver?

**Comment by calavera** [comment]
It should be 1.10, like @thaJeztah mentions. For good or bad, we agreed on keeping something deprecated for two releases.

**Comment by LK4D4** [comment]
@calavera What if there won't be replacement ready for 1.10? :)

**Comment by calavera** [comment]
@LK4D4 the current change says that it will be removed in 1.9, so I hope there is a replacement for 1.10. Besides, 1.10 gives us more flexibility in case of not having a replacement.

**Comment by thaJeztah** [comment]
@LK4D4 at that point we can decide to either remove it (without replacement), or keep it around for (one) more release.

My point was that 1 release (two months) notice might be too short for users.

**Comment by icecrime** [comment]
@LK4D4 I guess what it means is that user who rely on this should not upgrade to <whatever_version_drops_support>. We warn early enough for implementers to provide an alternative.

**Comment by thaJeztah** [comment]
@icecrime not sure I understand your opinion; you want to *remove* in 1.9, or 1.10?

**Comment by icecrime** [comment]
@thaJeztah I think it's 1.10 (two releases deprecation warning).

File: docs/misc/deprecated.md [hunk 4]
```diff
@@ -12,6 +12,14 @@ parent = "mn_use_docker"
 
 The following list of features are deprecated.
 
+### LXC built-in exec driver
+**Deprecated In Release: v1.8**
+
+**Target For Removal In Release: v1.9**
```
Comment (thaJeztah) [inline docs/misc/deprecated.md]: @crosbymichael could you change this to 1.10 (see the discussion above)?

**Comment by thaJeztah** [comment]
Thanks, LGTM

**Comment by calavera** [comment]
cherry picked into #15091

**Pull request merged**